### DISCUSSION TITLE
Double-handed slider

### DISCUSSION THREAD

**Discussion by majorsopa** [emilk/egui#2006] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/73245063/188519188-9b539648-f4b3-4336-a941-a5d72a32583f.png)

The image attached is a google image roughly showing what I am talking about. Instead of its value being an `f64`, its value would be a `RangeInclusive<f64>` or similar. How would I go about creating this slider with 2 handles?

**Comment by hacknus** [comment]
I actually started implementing something like this, but as a value I did not use `RangeInclusive<f64>`, I just used two bounds each as `f64`.
If somebody is interested I can put together a minimal working example. I used it below a small plot to set a window on the x-axis.

Though I have to say, my current implementation gets a bit buggy when the two values get close to each other, since there is a box around each element where we can grab it. When they overlap I had to set one as preferred such that you don't grab both. My current approach is that always the lower one is preferred so you have to move this one out of the way first if they come too close... this is _okay_ but by no means a nice user experience. Very much open to inputs about this.

**Reply by Mingun** [reply]
I think, that you could detect from what side the mouse pointer was approached and grab the corresponding boundary.

**Reply by hacknus** [reply]
@Mingun good idea, I'll try that. But first I need to clean up and make it more similar to the existing Slider widget.

**Comment by hacknus** [comment]
My first working example can be found [here](https://github.com/hacknus/egui/tree/double_range_slider/examples/double_range_slider). The code is not at all cleaned up, I'll try to put it in a more similar shape as the existing `Slider` widget.

**Reply by hacknus** [reply]
update: I don't think I'll be able to proceed here on my own. I understand way too little about the implementation of widgets/slider.. Could benefit from help.

**Comment by hacknus** [comment, marked as answer]
Published as a third party crate: [egui_double_slider](https://github.com/hacknus/egui_double_slider).

**Reply by majorsopa** [reply]
Awesome! I'm not gonna lie I kinda made my own implementation of it within a week of posting the original question and forgot to share it anywhere.
I hope this will lead the way to there being an implementation in the library itself.

**Reply by hacknus** [reply]
@majorsopa feel free to open a PR to this crate if you have any improvements from your version!

egui said the double slider was too niche for an implementation in the main library and suggested to publish it as a third party crate, see the PR here: #5332.
